tests/examples/test_ecephys_data_read.cpp

```diff
@@ -0,0 +1,215 @@
+{
+  SECTION("ecephys data read example")
+  {
+    // [example_read_mockdata_snippet]
+    // setup mock data for writing
+    SizeType numSamples = 100;
+    SizeType numChannels = 2;
+    std::vector<Types::ChannelVector> mockArrays = getMockChannelArrays();
+    BaseDataType dataType = BaseDataType::F32;
+    std::vector<std::string> mockChannelNames =
+        getMockChannelArrayNames("esdata");
+    std::vector<std::vector<float>> mockData =
+        getMockData2D(numSamples, numChannels);
+    std::vector<double> mockTimestamps = getMockTimestamps(numSamples, 1);
+    // To verify that the data was written correctly, we here transpose the
+    // mockData (which is per channel) to the (time x channel) layout used
+    // in the ElectricalSeries in the NWB file so we can compare
+    std::vector<std::vector<float>> mockDataTransposed;
+    mockDataTransposed.resize(numSamples);
+    for (SizeType s = 0; s < numSamples; s++) {
+      mockDataTransposed[s].resize(numChannels);
+      for (SizeType c = 0; c < numChannels; c++) {
+        mockDataTransposed[s][c] = mockData[c][s];
+      }
+    }
+    // [example_read_mockdata_snippet]
+
+    // [example_read_create_file_snippet]
+    // setup io object
+    std::string path = getTestFilePath("ElectricalSeriesReadExample.h5");
+    std::shared_ptr<BaseIO> io = createIO("HDF5", path);
+
+    // setup the NWBFile
+    NWB::NWBFile nwbfile(io);
+    nwbfile.initialize(generateUuid());
+
+    // create the RecordingContainer for managing recordings
+    std::unique_ptr<NWB::RecordingContainers> recordingContainers =
+        std::make_unique<NWB::RecordingContainers>();
+
+    // create a new ElectricalSeries
+    Status resultCreate = nwbfile.createElectricalSeries(
+        mockArrays, mockChannelNames, dataType, recordingContainers.get());
+    REQUIRE(resultCreate == Status::Success);
+
+    // get the new ElectricalSeries
+    NWB::ElectricalSeries* electricalSeries =
+        static_cast<NWB::ElectricalSeries*>(
+            recordingContainers->getContainer(0));
+    REQUIRE(electricalSeries != nullptr);
+
+    // start recording
+    Status resultStart = io->startRecording();
+    REQUIRE(resultStart == Status::Success);
+
+    // write channel data
+    for (SizeType ch = 0; ch < numChannels; ++ch) {
+      electricalSeries->writeChannel(
+          ch, numSamples, mockData[ch].data(), mockTimestamps.data());
+    }
+    io->flush();
+    // [example_read_create_file_snippet]
+
+    // [example_read_get_data_wrapper_snippet]
+    // Get a ReadDatasetWrapper<float> for lazy reading of ElectricalSeries.data
+    // By specifying the value type as a template parameter allows us to read
+    // typed data
+    auto readDataWrapper = electricalSeries->dataLazy<float>();
+    // [example_read_get_data_wrapper_snippet]
+
+    // [example_read_get_datablock_snippet]
+    // Read the full  ElectricalSeries.data back
+    DataBlock<float> dataValues = readDataWrapper->values();
```

Comment:
Are you refering to 1) ``electricalSeries->dataLazy<float>();`` or to 2) ``DataBlock<float> dataValues = readDataWrapper->values()``. For 1) we could set the default for ``readData`` in #91 to be float in `ElectricalSeries` so we don't need to provide the template. For 2) we could just say `auto dataValues` instead. 

Note, reading with a data type of std::any I don't think will work right now, because we can't simply cast ``std::vector<float>`` to  ``std::vector<std::any>``. I.e., when we don't specify the data type, we can only read as `DataBlockGeneric` but not as `DataBlock` right now.

Reply:
> For 1) we could set the default for readData in https://github.com/NeurodataWithoutBorders/aqnwb/pull/91 to be float in ElectricalSeries so we don't need to provide the template.

I was referring to `electricalSeries->dataLazy<float>();`.  I didn't connect that if we want to read with data type `std::any` then we would have to use `DataBlockGeneric`. 

> we could set the default for readData in https://github.com/NeurodataWithoutBorders/aqnwb/pull/91 to be float in ElectricalSeries so we don't need to provide the template

That would be helpful. It wasn't a huge block to understanding, but I think in general having exactly how the `ReadDataWrapper` is templated influencing the use of `DataBlock` or `DataBlockGeneric` might get tricky. We could discuss this related to the proposal to only use `DataBlock`